### PR TITLE
Apply forward token and payload based on API spec

### PR DESCRIPTION
**Pull request by diemtvu**
Config JWT filter for forwarding token and/or payload as specified by the API.

### CODE REVIEW

File: pilot/pkg/security/authn/v1beta1/policy_applier.go [hunk 1]
```diff
@@ -167,14 +167,17 @@ func convertToEnvoyJwtConfig(jwtRules []*v1beta1.JWTRule) *envoy_jwt.JwtAuthenti
...
-		// TODO(diemtvu): set forward based on input spec after https://github.com/istio/api/pull/1172
 		provider := &envoy_jwt.JwtProvider{
 			Issuer:            jwtRule.Issuer,
 			Audiences:         jwtRule.Audiences,
-			Forward:           false,
+			Forward:           jwtRule.ForwardOriginalToken,
 			PayloadInMetadata: jwtRule.Issuer,
 		}
 
+		if len(jwtRule.OutputPayloadToHeader) > 0 {
```
Comment (yangminzhu) [inline pilot/pkg/security/authn/v1beta1/policy_applier.go]: Does this check make any difference? Can we set the `OutputPayloadToHeader` in line 170 together.

Comment (diemtvu) [reply pilot/pkg/security/authn/v1beta1/policy_applier.go]: Good point.

File: pilot/pkg/security/authn/v1beta1/policy_applier_test.go [hunk 2]
```diff
@@ -363,6 +363,112 @@ func TestJwtFilter(t *testing.T) {
...
+			name: "Output payload",
+			in: []*model.Config{
+				{
+					Spec: &v1beta1.RequestAuthentication{
+						JwtRules: []*v1beta1.JWTRule{
+							{
+								Issuer:                "https://secret.foo.com",
+								JwksUri:               jwksURI,
+								ForwardOriginalToken:  true,
+								OutputPayloadToHeader: "x-foo",
```
Comment (incfly) [inline pilot/pkg/security/authn/v1beta1/policy_applier_test.go]: what's the behavior if two rules specify the same payload to header? the last wins? should we add a test for that?

**Comment by incfly** [comment]
not sure about echo, but `httpbin/headers` allow to inspect request header, client can thus indirectly verify server gets the jwt payload.

File: pilot/pkg/security/authn/v1beta1/policy_applier_test.go (hunk 2 above)
Comment (diemtvu) [reply pilot/pkg/security/authn/v1beta1/policy_applier_test.go]: Yes. On the other hand, we do not support multiple JWTs per request (yet). We will need to document it clearly in istio.io, but for now, I don't think that test case is necessary.

**Comment by diemtvu** [comment]
> for e2e not sure about echo, but `httpbin/headers` allow to inspect request header, client can thus indirectly verify server gets the jwt payload.

We are using `echo` for e2e test so I don't want to switch just for this. It is possible to parse the echo output, but is a bit complex so I want to do in different PR.

**Pull request merged**